src/main/java/com/kodilla/ecommercee/controller/UserController.java

```diff
@@ -12,30 +14,35 @@
 @CrossOrigin("*")
 public class UserController {
 
-    private UserRepository userRepository;
+    @Autowired
+    private UserMapper userMapper;
+
+    @Autowired
+    private UserService userService;
 
-    public UserController(UserRepository userRepository) {
-        this.userRepository = userRepository;
-    }
     @GetMapping(value = "getUser")
-    public User getUser(@RequestParam Long id) throws UserNotFoundException {
-        return userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+    public UserDto getUser(@RequestParam Long id) throws UserNotFoundException {
+        return userMapper.mapToUserDto(userService.getUser(id).orElseThrow(UserNotFoundException::new));
     }
+
     @GetMapping(value = "getUsers")
-    public List<User> getUsers() {
-        return userRepository.findAll();
+    public List<UserDto> getUsers() {
+        return userMapper.mapToUserDtoList(userService.getUsers());
     }
+
     @PostMapping(value = "addUser", consumes = "application/json")
-    public void addUser(@RequestBody User user) {
-        userRepository.save(user);
+    public UserDto addUser(@RequestBody UserDto userDto) {
+        return userMapper.mapToUserDto(userService.saveUser(userMapper.mapToUser(userDto)));
     }
+
     @PutMapping(value = "updateUser", consumes = "application/json")
-    public void updateUser(@RequestBody User user) {
-        userRepository.save(user);
+    public UserDto updateUser(@RequestBody UserDto userDto) {
+        return userMapper.mapToUserDto(userService.saveUser(userMapper.mapToUser(userDto)));
     }
+
     @DeleteMapping(value = "deleteUser")
     public void deleteUser(@RequestParam Long id) throws UserNotFoundException {
-        User user = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
-        userRepository.delete(user);
+        userMapper.mapToUserDto(userService.getUser(id).orElseThrow(UserNotFoundException::new));
```

Comment:
Nie korzystasz nigdzie z obiektu który wyszukujesz - przenieś to lepiej do serwisu - polecam jedną z funkcji Optional -> ifPresent i wtedy w else możesz rzucić wyjątkiem :)